snapcraft/tests/store/test_store_client.py

```diff
@@ -1183,3 +1185,16 @@ def test_sign_dev_agreement_exception_store_down(self):
             'There was an error while signing developer agreement.\n'
             'Reason: \'Internal Server Error\'\n'
             'Text: \'Broken\'')
+
+
+class UserAgentTestCase(tests.TestCase):
+
+    def test_user_agent(self):
+        arch = ProjectOptions().deb_arch
+        expected = 'snapcraft/{} {} ({})'.format(
+            snapcraft_version,
+            '/'.join(platform.dist()[0:2]),  # i.e. Ubuntu/16.04
+            arch,
+        )
+        actual = storeapi._get_user_agent()
+        self.assertEqual(expected, actual)
```

Comment:
ok, this is not very useful as it will be really difficult to make it fail. Using fakes instead of the real calls to deb_arch, version and dist would make it more readable, but using fakes is the downside.
Oh well, at least it will catch typos or when somebody tries to change the function.